### ISSUE TITLE
No other errors are reported when error in `for-loop &[T]`

### ISSUE THREAD

**Issue by reddevilmidzy** [rust-lang/rust#149446]
### Code

```Rust
#![allow(unused)]

fn main() {
    let x = 0;
    x += 1;

    for y in &[0u8, 1u8] {
        y += 1;
    }
}
```

### Current output

```Shell
error[E0368]: binary assignment operation `+=` cannot be applied to type `&u8`
 --> src\main.rs:6:9
  |
6 |         y += 1;
  |         -^^^^^
  |         |
  |         cannot use `+=` on type `&u8`

For more information about this error, try `rustc --explain E0368`.
```

### Desired output

```Shell
error[E0384]: cannot assign twice to immutable variable `x`
 --> src\main.rs:3:5
  |
2 |     let x = 0;
  |         - first assignment to `x`
3 |     x += 1;
  |     ^^^^^^ cannot assign twice to immutable variable
  |
help: consider making this binding mutable
  |
2 |     let mut x = 0;
  |         +++

error[E0368]: binary assignment operation `+=` cannot be applied to type `&u8`
 --> src\main.rs:6:9
  |
6 |         y += 1;
  |         -^^^^^
  |         |
  |         cannot use `+=` on type `&u8`
```

### Other cases

```Rust

```

### Rust Version

```Shell
rustc 1.93.0-nightly (cc3eee7fb 2025-11-28)
binary: rustc
commit-hash: cc3eee7fbe17ea4b7238531cb97e1b7b8bd6afce
commit-date: 2025-11-28
host: x86_64-pc-windows-msvc
release: 1.93.0-nightly
LLVM version: 21.1.5
```

**Comment by reddevilmidzy** [comment]
Ah, I understand. thanks
Should I close the issue?

**Comment by chenyukang** [comment]
yeah, it's Ok to close if you like.
